integration-tests/src/test/resources/setupenv.sh

```diff
@@ -226,61 +227,46 @@ if [ "$SHARED_CLUSTER" = "true" ]; then
   clean_shared_cluster
     
   if [ "$JRF_ENABLED" = true ] ; then
-	pull_tag_images_jrf	
+	pull_tag_images_jrf
   else
   	export IMAGE_PULL_SECRET_OPERATOR=$IMAGE_PULL_SECRET_OPERATOR
-	export IMAGE_PULL_SECRET_WEBLOGIC=$IMAGE_PULL_SECRET_WEBLOGIC
-  
+	export IMAGE_PULL_SECRET_WEBLOGIC=$IMAGE_PULL_SECRET_WEBLOGIC  
   
   	if [ "$IMAGE_PULL_POLICY_OPERATOR" = "Always" ]; then 
 		if [ -z "$REPO_REGISTRY" ] || [ -z "$REPO_USERNAME" ] || [ -z "$REPO_PASSWORD" ] || [ -z "$REPO_EMAIL" ]; then
 			echo "Provide Docker login details using REPO_REGISTRY, REPO_USERNAME, REPO_PASSWORD & REPO_EMAIL env variables to push the Operator image to the repository."
 			exit 1
 	  	fi
+
+                echo "login to $REPO_REGISTRY"
+                docker login $REPO_REGISTRY -u $REPO_USERNAME -p $REPO_PASSWORD
 	  	
 	  	echo "Creating Registry Secret"
 	  	kubectl create secret docker-registry $IMAGE_PULL_SECRET_OPERATOR  \
 		    --docker-server=$REPO_REGISTRY \
 		    --docker-username=$REPO_USERNAME \
 		    --docker-password=$REPO_PASSWORD \
-		    --docker-email=$REPO_EMAIL 
+		    --docker-email=$REPO_EMAIL  \
+                    --dry-run -o yaml | kubectl apply -f -
```

Comment:
what it this?

Reply:
This will create the secret and update it if already exists. without the kubectl apply it will not update if the secret already exists.

Reply:
update existing secret without deleting and recreating.